test/e2e/tls_util_test.go

```diff
@@ -197,6 +201,44 @@ func TestNoneOfCaAndAppSecretExist(t *testing.T) {
 	verifyCASecret(t, caSecret, namespace)
 }
 
+// TestCustomCA ensures that if a user provides a custom Key and Cert and the CA and Application TLS assets
+// do not exist, the GenerateCert method can use the custom CA to generate the TLS assest.
+func TestCustomCA(t *testing.T) {
+	f := framework.Global
+	ctx := f.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cg := tlsutil.NewSDKCertGenerator(f.KubeClient)
+
+	customConfig := &tlsutil.CertConfig{
+		CertName: certName,
+		CAKey:    "testdata/ca.key",
+		CACert:   "testdata/ca.crt",
+	}
+	appSecret, _, _, err := cg.GenerateCert(newDummyCR(namespace), newAppSvc(namespace), customConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	verifyAppSecret(t, appSecret, namespace)
+
+	// ensure caConfigMap does not exist in k8s cluster.
+	_, err = framework.Global.KubeClient.CoreV1().Secrets(namespace).Get(caConfigMapAndSecretName, metav1.GetOptions{})
+	if !strings.Contains(err.Error(), "not found") {
```

Comment:
you can use functions within the k8s errors pkg to check if the err is a `not found` error. e.g

```go
import apiErrors "k8s.io/apimachinery/pkg/api/errors"

func TestCustomCA(t *testing.T) {
  ...
  if !apiErrors.IsNotFound(err) {
    t.Fatal(err)
 }
}
``